docs/devel/infrastructure-interaction.md

```diff
@@ -0,0 +1,85 @@
+
+**NFS Server Storage Example**
+
+As an example lets say I want to use an NFS server to expose 10 exports for pod storage. I believe that either an admin or the infrastructure itself should make all of the information needed to mount those 10 exports available to the API server. I think this should be done as a new ‘volume’ resource (yes, very like the volume portion of the pod definition).  So (for NFS) the cloud provider (or maybe a kube admin) would make available the servername, export name, and any credentials necessary to use the mount. Now the kube admin/operator/user should be able to define a pod which references the storage by name. He should NOT have to know all of the storage details (as the volume requires today). The apiserver should take the volumeMount from the pod definition, the volume resource, and make all of that available to the kubelet. The dev who wrote the pod was told to use 'eparis.volume.42' and that's all they know. The kublet however gets nfs-server.example.com:/export/42/eparis.volume.stage/ along with a kerberos keytab from the apiserver. It can now mount and use the NFS mount.  I believe the concept should be able to map pretty cleanly to GCE PDs, or any storage concept (durable/empty/host are a little special, but not terribly outside the model.)
+
+Openshift or others are certainly going to want to write a high level tool, which attempts to automatically create NFS exports or purchase more storage from GCE when requested. Thus imitating by doing work at a higher level the other infrastructure kube interaction perspective.  But the clean separation makes that possible rapidly.  That tool should interact initially with the infrastructure and the API server, to make the volume resource.  Although eventually I’m sure they’d like to choose to interact with one or the other.
+
+Separating the volume to a first class resource instead of part of the pod definition gets you all sorts of cool stuff.  Couple of initial thoughts are that pod definitions become much more portable, since the infrastructure specific portions are abstracted out.  Access controls become much more reasonable, as you can easily do checks on creating the volume resource and between volume resources and pods.  More thoughts below...
```

Comment:
Definitely for remote volumes this makes sense. It's also similar to what was discussed for "durable" local volumes with lifetimes independent from pods.

Mapping in a host directory should be a privileged operation, and is sort of a special case.

How to populate volumes with data (git, URL, docker image, etc.) is a separate issue -- that should be handled by invoking a container.